### PR TITLE
[FIX] web: domains evaluation context

### PR DESCRIPTION
**Pull request by Feyensv**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/views/fields/relational_utils.js [hunk 1]
```diff
@@ -213,6 +213,8 @@ export class Many2XAutocomplete extends Component {
     }
 
     async loadOptionsSource(request) {
+        // TODO if request is empty string, call search directly to avoid useless
```
Comment (kebeclibre) [inline addons/web/static/src/views/fields/relational_utils.js:216]: nope, the return of name_search is the name_get on the results. Besides, python side name_search does what you say

File: addons/web/static/src/views/relational_model.js [hunk 2]
```diff
@@ -745,7 +745,10 @@ export class Record extends DataPoint {
             this.activeFields[fieldName].domain,
         ];
 
-        const evalContext = this.evalContext;
+        const evalContext = Object.assign(
```
Comment (kebeclibre) [inline addons/web/static/src/views/relational_model.js:748]: I don't think you wanna overwrite this.context
I guess ```
Object.assign({}, this.context, this.evalContext);
``` is more correct

File: addons/web/static/src/views/fields/relational_utils.js (hunk 1 above)
Comment (Feyensv) [reply addons/web/static/src/views/fields/relational_utils.js:216]: Indeed, stupid remark :D

File: addons/web/static/src/views/relational_model.js (hunk 2 above)
Comment (Feyensv) [reply addons/web/static/src/views/relational_model.js]: Done :+1:
I'll leave it to you to add a good test covering the problem ? :pray:

**Comment by kebeclibre** [comment]
https://github.com/odoo/odoo/pull/103333

**Pull request closed without merging**